### PR TITLE
addGitignore&RemoveOldMetadataFiles&AddSbtEclipseSupport

### PR DESCRIPTION
**Pull request by rayeaster**
This Pull Request contains 3 commits:

- add sbt eclipse support for scala-ide(eclipse), reference https://medium.com/@murphye/up-and-running-using-sbt-with-eclipse-or-intellij-640bb121e128
- add more file suffix to ignore in git like .project, .classpath and .cache-main
- remove old project/classpath files

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -50,7 +50,9 @@ lazy val commonSettings = Seq(
           <url>http://haifengl.github.io/</url>
         </developer>
       </developers>
-  )
+  ),
+  EclipseKeys.withSource := true,
```
Comment (haifengl) [inline build.sbt]: What do these two lines mean?

File: .gitignore [hunk 2]
```diff
@@ -21,3 +21,9 @@ hs_err_pid*
 *.iml
 **/target
 
+.project
+.settings
+.cache-main
+.cache-tests
+.classpath
+*/.gitignore
```
Comment (haifengl) [inline .gitignore]: why */.gitignore?

File: build.sbt (hunk 1 above)
Comment (rayeaster) [reply build.sbt]: https://github.com/sbt/sbteclipse/wiki/Using-sbteclipse#withsource
https://github.com/sbt/sbteclipse/wiki/Using-sbteclipse#withjavadoc
basically, these two allows to download source and java doc for convenience reference

Comment (rayeaster) [reply build.sbt]: https://github.com/sbt/sbteclipse/wiki/Using-sbteclipse#withsource
https://github.com/sbt/sbteclipse/wiki/Using-sbteclipse#withjavadoc
basically, these two allows to download source and java doc for convenience reference

File: .gitignore (hunk 2 above)
Comment (rayeaster) [reply .gitignore]: it seems to me that there is no need for a separate gitignore file under each sub folder/project so I add this line. If that is not true or you want more flexibility, of course we could remove this line

**Comment by haifengl** [comment]
Thanks

**Comment by rayeaster** [comment]
My pleasure. Thanks for taking to review.

**Pull request merged**